packages/hierarchies/learning/Formatting.md

```diff
@@ -0,0 +1,221 @@
+
+<!-- [[include: [Presentation.Hierarchies.Formatting.NodeLabelFormattingExamples.Imports, Presentation.Hierarchies.Formatting.InstanceNodesQueryDefinitionLabelFormattingExample], ts]] -->
+<!-- BEGIN EXTRACTION -->
+
+```ts
+import { createHierarchyProvider, createNodesQueryClauseFactory } from "@itwin/presentation-hierarchies";
+import { createDefaultInstanceLabelSelectClauseFactory, ECSql } from "@itwin/presentation-shared";
+
+const hierarchyProvider = createHierarchyProvider({
+  imodelAccess,
+  hierarchyDefinition: {
+    defineHierarchyLevel: async () => [
+      // the hierarchy definition returns BisCore.SpatialCategory nodes
+      {
+        fullClassName: "BisCore.SpatialCategory",
+        query: {
+          ecsql: `
+            SELECT ${await createNodesQueryClauseFactory({ imodelAccess }).createSelectClause({
+              ecClassId: { selector: "this.ECClassId" },
+              ecInstanceId: { selector: "this.ECInstanceId" },
+              // generally, one of the `IInstanceLabelSelectClauseFactory` implementations, delivered with `@itwin/presentation-shared` package, should be used,
+              // but for demonstration purposes, a custom implementation is used here
+              nodeLabel: {
+                selector: ECSql.createConcatenatedValueJsonSelector([
+                  // create a selector for `CodeValue` property value
+                  { propertyClassName: "BisCore.SpatialCategory", propertyClassAlias: "this", propertyName: "CodeValue" },
+                  // include a static string value
+                  { type: "String", value: " [" },
+                  // create a selector for `ECInstanceId` property value in hex format
+                  { selector: `printf('0x%x', ${ECSql.createRawPropertyValueSelector("this", "ECInstanceId")})` },
+                  // include a static string value
+                  { type: "String", value: "]" },
+                ]),
+              },
+            })}
+            FROM BisCore.SpatialCategory this
+          `,
+        },
+      },
+    ],
+  },
+});
+
+// all returned nodes have their labels set in format "{CodeValue} [{ECInstanceId}]"
+const nodes = hierarchyProvider.getNodes({ parentNode: undefined });
+```
+
+<!-- END EXTRACTION -->
+
+Finally, in case 3, the grouping node's label is formatted automatically based on the type of grouping:
+
+- In case of class grouping, the labels is the class' display label and needs to extra formatting.
+- In case of label grouping, the label is taken from grouped nodes whose labels are already formatted.
+- In case of property grouping, the label is built using formatted property values. The formatter is applied to each property value separately using formatter used by `HierarchyProvider`.
```

Comment:
```suggestion
- In case of property grouping, the label is built using formatted property values. The formatter used by `HierarchyProvider` is applied to each property value separately`.
```